app/src/main/java/com/gdgevents/gdgeventsapp/common/navigation/Destination.kt

```diff
@@ -1,5 +1,24 @@
 package com.gdgevents.gdgeventsapp.common.navigation
 
+import kotlinx.serialization.Serializable
+
 sealed interface Destination {
+    val route: String
+
     // add Destinations
-}
\ No newline at end of file
+
+    @Serializable
+    data object OnboardingScreen : Destination {
+        override val route: String = "onboardingScreen"
+    }
+    @Serializable
+    data object EnterLocationScreen : Destination {
+        override val route: String = "enterLocationScreen"
+    }
+
+    @Serializable
```

Comment:
Why this annotation?

Reply:
It is required for the route to be annotated with `@Serializable` in the type safe navigation
it should look like this
```
sealed interface Destination {

    @Serializable
    data object HomeRoute: Destination

    @Serializable
    data class DetailRoute(val id: String): Destination
}
```
```
@Composable
fun GdgNavHost(
    modifier: Modifier = Modifier
) {
    val navController = rememberNavController()
    NavHost(
        navController = navController,
        startDestination = HomeRoute,
        modifier = modifier
    ) {
        composable<HomeRoute> {
            HomeScreen(
                goToDetails = { id ->
                    val detailRoute = DetailRoute(id)
                    navController.navigate(detailRoute)
                }
            )
        }
        composable<DetailRoute> { backStackEntry ->
            val detailRoute = backStackEntry.toRoute<DetailRoute>()
            DetailsScreen(
                id = detailRoute.id
            )
        }
    }
}
```

Reply:
The type safety is for passing arguments between screens not the routes